GoogleForCreators/web-stories-wp
# Design System Storybook: Snackbar

## Summary

Adds UI for snackbar for use in dashboard and editor[ based on figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=1533%3A260040)

(light mode)
![Screen Shot 2020-11-13 at 12 04 17 PM](https://user-images.githubusercontent.com/10720454/99111059-a8b37a00-25a8-11eb-8309-633ae0ff57e2.png)
(dark mode)
![Screen Shot 2020-11-13 at 12 04 11 PM](https://user-images.githubusercontent.com/10720454/99111062-a9e4a700-25a8-11eb-964c-45319b8efc66.png)



## Relevant Technical Choices

- Since the dashboard and editor have different implementations of snackbars I thought it best to not force an update of functionality and just supply a reskin. We have a container to hold the snackbar contents and a component to display contents set to auto dismiss unless specified not too.
- This week the designs changed to invert the colors used in the snackbars - so if you are in light mode your snackbar is in dark theme color palette and if you are in dark mode your snackbar is in the light theme color palette. ~There's one discrepancy here - the link color shown IN the snackbar using light theme colors on figma isn't updated to the finalized color palette. I am using the link color appropriately from the theme and am making [note of the discrepancy in the ticket](https://github.com/google/web-stories-wp/issues/4957#issuecomment-726990942) so that when this gets to UAT it can be confirmed.~ Resolved - link color for light theme is dropped to blue 50.
- This will allow for a pretty easy replacement, like so: 
![Screen Shot 2020-11-06 at 2 04 41 PM](https://user-images.githubusercontent.com/10720454/98414735-23c0e180-2039-11eb-9dbc-9066d278e93d.png)

- Noticed that constants could have their own subdirectory in the theme for easier organization, so I did that here since other component PRs in design system are merged. Bulk of these files are just cleaning up paths. 
  - Nesting `TYPOGRAPHY_` in `TYPOGRAPHY` and dropping the typography prefix
  - Moving wordPress constants from dialog to constants/ in same structure as typography (`WP_ADMIN.`) 


## To-do

[Verify the following with Sam: ](https://github.com/google/web-stories-wp/issues/4957#issuecomment-723301368)
(verified: https://github.com/google/web-stories-wp/issues/4957#issuecomment-723783024 )
~1. we shouldn't have a dismiss button 
2. we always want these center - bottom aligned
3. what happens when they stack?~


## User-facing changes

- none, just storybook

## Testing Instructions

- Look at designSystem/components/snackbar locally and see the snackbar
- Verify tests run locally if you want

---



Addresses #4957
